common/src/main/java/org/esbtools/message/admin/common/EsbMessageAdminServiceImpl.java

```diff
@@ -196,6 +210,80 @@ public void create(EsbMessage esbMessage, Map<String, List<String>> extractedHea
         entityMgr.persist(eme);
     }
 
+    public MetadataResponse resubmitMessage(EsbMessage esbMessage) {
+
+        ResubmitRequest request = new ResubmitRequest();
+
+        EsbMessageEntity persistedMessage = entityMgr.find(EsbMessageEntity.class, esbMessage.getId());
+
+        request.setHeaders( reduceToEsbHeaders(persistedMessage) );
+        request.setDestination( persistedMessage.getHeader("esbResubmitDestination").getValue() );
+        request.setSystem( persistedMessage.getSourceSystem() );
+
+        // scold the user for trying to update instead of insert
+        if( persistedMessage == null ){
+            throw new IllegalArgumentException("Message {\n}"+esbMessage.toString()+"\n} does not exist in backend store, cannot update.");
+        }
+
+        // explicitly check if the loaded message is in our list of allowed message types
+        if( isEditableMessage(esbMessage) ){
+            request.setPayload( esbMessage.getPayload() );
+        } else {
+            request .setPayload( persistedMessage.getPayload() );
+        }
+        saveAuditEvent( new AuditEvent(DEFAULT_USER, RESUBMIT_EVENT, "", "", "", request.getPayload().toString() ) );
+        MetadataResponse result = sendMessageToResubmitGateway( request.toString() );
+        persistedMessage.setResubmittedOn(new Date());
+        entityMgr.flush();
+        return result;
+    }
+
+    private MetadataResponse sendMessageToResubmitGateway( String message ) {
+
+        if( sendMessageToRESTEndPoint( message, getResubmitRestEndpoints() ) ){
+            return new MetadataResponse();
+        } else{
+            MetadataResponse result = new MetadataResponse();
+            result.setErrorMessage("Unable to resubmit message.");
+            return result;
+        }
+
+    }
+
+    private Boolean sendMessageToRESTEndPoint( String message, List<String> endpoints ) {
```

Comment:
Small nitpick about using REST in the method name instead of Rest (i.e. sendMessageToRestEndpoint instead)